scripts/analytics/logs_analytics.py

```diff
@@ -0,0 +1,120 @@
+from datetime import datetime as dt
+from datetime import timedelta
+import matplotlib.pyplot as plt
+
+from modularodm import Q
+
+from website.app import init_app
+from website.models import NodeLog, User, Node
+
+color_map = {
+    'comments': 'red',
+    'wiki': 'skyblue',
+    'registrations': 'lime',
+    'nodes': 'yellow',
+    'files': 'magenta'
+}
+
+NUMBER_OF_DAY_DATA_POINTS = 30
+NUMBER_OF_USERS_TO_SAMPLE = 1
+
+def recent_time_frame(days=NUMBER_OF_DAY_DATA_POINTS):
+    now = dt.utcnow()
+    return [now - timedelta(days=i) for i in range(0, days)]
+
+def build_time_query(end):
+    return Q('date', 'gt', end - timedelta(days=1)) & Q('date', 'lt', end)
+
+def order_users_get(sample_size=NUMBER_OF_USERS_TO_SAMPLE):
+    print('Begin finding users.')
+    users = list(User.find())
+    for user in users:
+        how_many = len(list(NodeLog.find(Q('user', 'eq', user))))
+        if how_many < 1:
+            users.remove(user)
+    ordered = sorted(users, key=lambda x: x.get_activity_points(), reverse=True)
+    print('Found users.')
+    l = len(ordered)
+    return ordered[int(l/50): int(l/50) + sample_size], ordered[int(l/40): int(l/40) + sample_size], ordered[int(l/30): int(l/30) + sample_size], ordered[int(l/20): int(l/20) + sample_size]
+
+def get_agg_for_user(user, date):
+    ret = {}
+    nodes = list(Node.find(Q('contributors', 'contains', user._id)))
+    for node in nodes:
+        query_aggs = {
+            'comments': len(list(NodeLog.find(Q('was_connected_to', 'eq', node) & Q('action', 'eq', NodeLog.COMMENT_ADDED) & build_time_query(date)))),
```

Comment:
What are you trying to get here? Are you sure `was_connected_to` is correct? IIRC, this field was added before a log migration just to maintain provenance--it doesn't actually get used on the site.